### PR TITLE
chore: Use berty/mobile with Swift fix. Makefile: Use gobind cache.

### PR DESCRIPTION
**Pull request by jefft0**
Similar to https://github.com/berty/berty/pull/4404, use berty/mobile branch with fixes including no crash for Objective-C null return. Update the Makefile similar to Berty to use a cache for gomobile bind.

### CODE REVIEW

File: packages/Makefile [hunk 1]
```diff
@@ -81,6 +80,11 @@ IOS_DOC_DIR = $(DOC_DIR)/ios
 
 .PHONY: help build_core build_core.android build_core.ios build_demo build_demo.android build_demo.ios clean clean.android clean.ios docgen docgen.android docgen.ios fail_on_dev publish publish_bridge publish_bridge.android publish_bridge.ios publish_core publish_core.android publish_core.ios publish_demo publish_demo.android publish_demo.ios re re.android re.ios test test_bridge test_bridge.android test_bridge.ios test_core
 
+## Internal variables
+
+android_gomobile_cache="$(PWD)/android/.gomobile-cache"
+ios_gomobile_cache="$(PWD)/ios/.gomobile-cache"
```
Comment (aeddi) [inline packages/Makefile:86]:
```suggestion
android_gomobile_cache="$(ANDROID_BUILD_DIR_INT_CORE)/.gomobile-cache"
ios_gomobile_cache="$(IOS_BUILD_DIR_INT_CORE)/.gomobile-cache"
```

Comment (aeddi) [reply packages/Makefile:86]: Can you also remove the "## Internal variables" comment, use only uppercase for both variables and put them under the `IOS_BUILD_DIR_INT_CORE` and `ANDROID_BUILD_DIR_INT_CORE` lines?

File: packages/Makefile [hunk 2]
```diff
@@ -105,11 +109,21 @@ $(ANDROID_BUILD_DIR_MAV_CORE): $(ANDROID_CORE) $(MANIFEST_FILE)
...
 	@echo '------------------------------------'
-	$(GOMOBILE) bind -v $(GOMOBILE_OPT) -target=android$(GOMOBILE_TARGET) -o $(ANDROID_CORE) $(CORE_PACKAGE) $(EXT_PACKAGE)
+	go mod download
+	go run golang.org/x/mobile/cmd/gomobile init
+	mkdir -p $(android_gomobile_cache) android/libs
+	GO111MODULE=on go run golang.org/x/mobile/cmd/gomobile bind \
+		-o $(ANDROID_CORE) \
+		-v $(GOMOBILE_OPT) \
+		-cache $(android_gomobile_cache) \
+		-target=android$(GOMOBILE_TARGET) \
```
Comment (aeddi) [inline packages/Makefile:123]: Can you copy this from Berty:
```
GOMOBILE_IOS_TARGET ?= ios
GOMOBILE_ANDROID_TARGET ?= android
```
and modify the rules using it accordingly?

File: .gitignore [hunk 3]
```diff
@@ -13,3 +13,8 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+# Cache
+packages/android
+packages/ios
```
Comment (aeddi) [inline .gitignore:19]: Can you remove this? (see Makefile comments)

**Review by aeddi — CHANGES_REQUESTED** [review]

File: packages/Makefile (hunk 1 above)
Comment (jefft0) [reply packages/Makefile]: Done.

File: packages/Makefile (hunk 2 above)
Comment (jefft0) [reply packages/Makefile]: Done.

File: .gitignore (hunk 3 above)
Comment (jefft0) [reply .gitignore]: Done.

**Pull request merged**